src/content/en/showcase/2017/with.in.md

```diff
@@ -0,0 +1,331 @@
+
+[text-bitmap.js ShaderMaterial](https://vr.with.in/archive/text-sdf-bitmap/text-bitmap.js)
+
+Adjust the **afwidth** and **threshold** sliders on [vr.with.in/archive/text-sdf-bitmap/](https://vr.with.in/archive/text-sdf-bitmap/) to see the affect of the signed distance field shader.
+
+## 5. Usage
+
+
+
+For convenience, I created a TextBitmap wrapper class for the browserified three-bmfont-text
+
+
+[vr.with.in/archive/text-sdf-bitmap/](https://vr.with.in/archive/text-sdf-bitmap/)
+
+<figure>
+  <img src="/web/showcase/2017/images/with.in/7.png">
+  <figcaption>Text-sdf-bitmap in action</figcaption>
+</figure>
+
+
+```
+<script src="three-bmfont-text-bundle.js"></script>
+<script src="sdf-shader.js"></script>
+<script src="text-bitmap.js"></script>
+```
+
+
+Create an XHR request for the .json font file and create text object in the callback:
+
+
+```
+var bmtext = new TextBitmap({ options });
+```
+
+
+To change text:
+
+
+```
+bmtext.text = 'The quick brown fox jumps over the lazy dog.';
+
+scene.add( bmtext.group );
+hitBoxes.push( bmtext.hitBox );
+```
+
+
+The bitmap font's .png is loaded with THREE.TextureLoader in text-bitmap.js
+
+TextBitmap also includes an invisible hitbox for three.js raycast interaction, through a mouse, camera, or hand tracked motion controllers like Oculus Touch or the Vive controllers. The hitbox's size auto updates when you change the text options.
+
+Bmtext.group is added to the three.js scene. If you need to access the children / Object3D's, the scene graph for the text looks like:
+
+<img src="/web/showcase/2017/images/with.in/diagram2.png">
+
+## 6. Unminify json and modify xoffsets
+
+<img src="/web/showcase/2017/images/with.in/8.gif">
+
+
+If your kernings look off, you may need to edit the xoffsets in the json. Paste the json into [Jsbeautifier.org](http://jsbeautifier.org/)** **to get an unminified version of the file.
+
+The xoffset is essentially global kerning for one character. Kerning is for two specific characters that appear next to each other. The default values in the kernings array don't actually make a difference, and it would be too tedious to edit, so you can empty that array to decrease the file size of the json. Then edit the xoffsets for kerning. But first you'll have to figure out which characters go with which char id in the json.
+
+In [three-bmfont-text-bundle.js](https://vr.with.in/archive/text-sdf-bitmap/three-bmfont-text-bundle.js), insert a console.log after line 240:
+
+
+```
+var id = text.charCodeAt(i)
+// console.log(id);
+```
+
+
+Then type into dat.gui text field on https://vr.with.in/archive/text-[sdf-bitmap/](https://vr.with.in/archive/text-sdf-bitmap/) and check the console to find the corresponding id of a character.
+
+For example, in our bitmap font, "j" is consistently too far to the right. Its char id is 106. So find "id": 106 in the json and change its xoffset from -1 to -10.
+
+[animated gif of pre xoffset edits and post xoffset edits]
```

Comment:
This is in the source doc, I assume that the original author left it as a note to themselves. Removed.